component/src/main/java/org/wso2/siddhi/runner/beam/extension/BeamGroupByKeyProcessor.java

```diff
@@ -0,0 +1,159 @@
+
+        ComplexEventChunk<StreamEvent> complexEventChunk = new ComplexEventChunk<>(false);
+        //TODO use lock instead
+        //TODO check thread safety
+        Map<K, List<V>> groupByKeyMap = new HashMap<>();
+        synchronized (this) {
+            try {
+                while (streamEventChunk.hasNext()) {
+                    StreamEvent event = streamEventChunk.next();
+                    WindowedValue windowedValue = (WindowedValue) this.eventExecutor.execute(event);
+                    KV keyValueElement = (KV) windowedValue.getValue();
+                    if (groupByKeyMap.containsKey(keyValueElement.getKey())) {
+                        List<V> items = groupByKeyMap.get(keyValueElement.getKey());
+                        items.add((V) keyValueElement.getValue());
+                        groupByKeyMap.put((K) keyValueElement.getKey(), items);
+                    } else {
+                        List<V> item = new LinkedList<>();
+                        item.add((V) keyValueElement.getValue());
+                        groupByKeyMap.put((K) keyValueElement.getKey(), item);
+                    }
+                }
+                for (Map.Entry map: groupByKeyMap.entrySet()) {
+                    K key = (K) map.getKey();
+                    List<V> value = (LinkedList<V>) map.getValue();
+                    KV kv = KV.of(key, value);
+                    StreamEvent streamEvent = new StreamEvent(0, 0, 1);
+                    streamEvent.setOutputData(WindowedValue.valueInGlobalWindow(kv), 0);
+                    complexEventChunk.add(streamEvent);
+                }
+            } catch (Exception e) {
+                log.error(e.getMessage());
+            }
+        }
+        nextProcessor.process(complexEventChunk);
+
+    }
+
+
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                   SiddhiAppContext siddhiAppContext) {
+
+        //TODO get length of attributeExpressionExecutors
+        //TODO check instance type(variable or constant)
+        if (attributeExpressionLength != 1) {
+            throw new SiddhiAppCreationException(
+                    "Only one parameter can be specified for BeamGroupByKeyProcessor but "
+                            + attributeExpressionLength
+                            + " were given"
+            );
+        }
+
+        if (attributeExpressionExecutors[0].getReturnType() == Attribute.Type.OBJECT) {
+            this.eventExecutor = attributeExpressionExecutors[0];
+        } else {
+            throw new SiddhiAppCreationException(
+                    "First parameter should be of type Object but "
+                    + attributeExpressionExecutors[0].getReturnType().toString()
+                    + " was given"
+            );
+        }
+        return new LinkedList<>();
```

Comment:
here you are returning a empty LinkedList always. is this correct ?

Reply:
Yes else it returns an error when null is returned